common/test/acceptance/pages/studio/html_component_editor.py

```diff
@@ -7,25 +7,59 @@ class HtmlComponentEditorView(ComponentEditorView):
     """
     Represents the rendered view of an HTML component editor.
     """
-    def set_content_and_save(self, content):
-        """
-        Types content into the html component and presses Save.
+    def set_content_and_save(self, content, raw=False):
+        """Types content into the html component and presses Save.
+
+        Arguments:
+            content (str): The content to be used.
+            raw (bool): If true, edits in 'raw HTML' mode.
         """
-        self.set_content(content)
+        if raw:
+            self.set_raw_content(content)
+        else:
+            self.set_content(content)
+
         self.save()
 
-    def set_content_and_cancel(self, content):
-        """
-        Types content into the html component and presses Cancel to abort the change.
+    def set_content_and_cancel(self, content, raw=False):
+        """Types content into the html component and presses Cancel to abort.
+
+        Arguments:
+            content (str): The content to be used.
+            raw (bool): If true, edits in 'raw HTML' mode.
         """
-        self.set_content(content)
+        if raw:
+            self.set_raw_content(content)
+        else:
+            self.set_content(content)
+
         self.cancel()
 
     def set_content(self, content):
-        """
-        Types content into the html component, leaving the component open.
+        """Types content into the html component, leaving the component open.
+
+        Arguments:
+            content (str): The content to be used.
         """
         self.q(css='.edit-xblock-modal .editor-modes .editor-button').click()
-        editor = self.q(css=self._bounded_selector('.html-editor .mce-edit-area'))[0]
+
+        selector = '.html-editor .mce-edit-area'
+        editor = self.q(css=self._bounded_selector(selector))[0]
         ActionChains(self.browser).click(editor).\
-            send_keys([Keys.CONTROL, 'a']).key_up(Keys.CONTROL).send_keys(content).perform()
+            send_keys([Keys.CONTROL, 'a']).key_up(Keys.CONTROL).\
+            send_keys(content).perform()
+
+    def set_raw_content(self, content):
+        """Types content in raw html mode, leaving the component open.
+
+        Arguments:
+            content (str): The content to be used.
+        """
+        self.q(css='.edit-xblock-modal .editor-modes .editor-button').click()
```

Comment:
Haha, I actually had just one `set_content` with a raw boolean earlier, but @andy-armstrong recommended splitting them up: https://github.com/edx/edx-platform/blob/57d4b94930fb60d4693010a451c90a6d172e08dc/common/test/acceptance/pages/studio/html_component_editor.py#L24

The other key difference is that in the non-raw case, we find the `editor` element to click on as part of the action chain. After all of that, the only line that's the same in both cases is the first click statement. I will get that css into a variable though.